source/elements/oneMKL/source/domains/blas/imatcopy_batch.rst

```diff
@@ -0,0 +1,507 @@
+.. SPDX-FileCopyrightText: 2022 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_blas_imatcopy_batch:
+
+imatcopy_batch
+==============
+
+Computes a group of in-place scaled matrix transpose or copy operations
+using general matrices.
+
+.. _onemkl_blas_imatcopy_batch_description:
+
+.. rubric:: Description
+
+The ``imatcopy_batch`` routines perform a series of in-place scaled matrix
+copies or transpositions. They are batched versions of :ref:`onemkl_blas_imatcopy`,
+but the ``imatcopy_batch`` routines perform their operations with
+groups of matrices. Each group contains matrices with the same parameters.
+
+There is a *strided API*, in which the matrices in a batch are a set
+distance away from each other in memory and in which all matrices
+share the same parameters (for example matrix size), and a more
+flexible *group API* where each group of matrices has the same
+parameters but the user may provide multiple groups that have
+different parameters.
```

Comment:
Below it looks like you have a buffer api that is strided, and two USM apis -- strided and group.  Should buffer also ahve the group api?  or is there a reason why you don't provide it?  If there is a reason, it probably makes sense to add that discussion to the spec as well...

Reply:
The group API needs an array of pointers (ie `float `**`) and my understanding is sycl buffers do not handle this, so we do not support the group API with buffers. This is true for all our batched routines, eg [gemm](https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/blas/gemm.html#gemm-buffer-version).

We could certainly provide some reasoning for that here, but then would we also add it for the other batched routines? What language do you suggest?

Reply:
I see, would an array of sycl::buffers would be possible ?  I suppose you have discussed things internally within blas team so maybe there are reasons that wouldn't work ? It seems like sycl::buffer * is a meaningful object and  sycl::buffer ** is also reasonable, but it somewhat violates the C++ style of sycl objects,  maybe a std::vector<sycl::buffer> & could be better ??  :)

Barring any additions, it would make sense to say something like 

`The Group API argument structure is more natural when using USM pointers than sycl::buffers, so we only implement it for USM inputs.  On the other hand, the Strided API works for both cases and so USM and sycl::buffer APIs are provided.`

Reply:
@lhuot Can you comment on the group API buffer limitation?

I'll add something like what you propose to the routines added in this PR, but won't touch any of the existing documents.

Reply:
You cannot create vector or array of buffer accessor so that mean if you had such batch group API with buffer input you won't be able to implement real batch optimization with kernel accessing several matrices.